android/src/main/kotlin/io/customer/customer_io/CustomerIoPlugin.kt

```diff
@@ -147,23 +173,29 @@ class CustomerIoPlugin : FlutterPlugin, MethodCallHandler {
         val apiKey = configData.getString(Keys.Environment.API_KEY)
         val region = configData.getProperty<String>(
             Keys.Environment.REGION
-        )?.takeIfNotBlank().toRegion()
+        )?.takeIfNotBlank()
         val enableInApp = configData.getProperty<Boolean>(
             Keys.Environment.ENABLE_IN_APP
         )
 
         CustomerIO.Builder(
             siteId = siteId,
             apiKey = apiKey,
-            region = region,
+            region = Region.getRegion(region),
             appContext = application,
+            config = configData
         ).apply {
-            setClient(client = getUserAgentClient(packageConfig = configData))
-            setupConfig(configData)
             addCustomerIOModule(module = configureModuleMessagingPushFCM(configData))
             if (enableInApp == true) {
                 addCustomerIOModule(
-                    module = ModuleMessagingInApp()
+                    module = ModuleMessagingInApp(
+                        config = MessagingInAppModuleConfig.Builder()
+                            .setEventListener(CustomerIOInAppEventListener { method, args ->
+                                this@CustomerIoPlugin.activity?.runOnUiThread {
```

Comment:
This is the main reason why I bring up the strong reference to the Activity and a memory leak. The activity is used inside of the event listener object. 

So, if the CIO SDK keeps a strong reference to the in-app event listener --> the event listener keeps a strong reference to the Activity -> the Activity might not get freed from memory in this class? 

I am unsure if this is true?